x/staking/client/cli/tx.go

```diff
@@ -546,3 +542,15 @@ func BuildCreateValidatorMsg(clientCtx client.Context, config TxCreateValidatorC
 
 	return txBldr, msg, nil
 }
+
+func parseMinSelfDelegation(minSelfDelegationStr, expectedDenom string) (*sdk.Coin, error) {
```

Comment:
I do not think this function adds to the simplicity of the CLI flow, actually it makes it more confusing for me to read. I would suggest just parsing the value where you need it.

Reply:
1b6bcc2: applied your suggestion 👍